src/hti/re_dash.cljd

```diff
@@ -382,6 +375,40 @@
                         @flows)))
     nil))
 
+(defn- flows-match?
+  [flow1 flow2]
+  #(= (:id flow1) (:id flow2)))
+
+(defn- sort-flows
+  [new-flows]
+
+  ;; TODO - Dependency graph, see flow<-
+
+  new-flows)
+
+(defn reg-flow
+
+  ([flow]
+   (when-not (:id flow) (throw (ex-info ":id is mandatory" flow)))
+   (reg-flow (:id flow) flow))
+
+  ([flow-id {:keys [inputs output path] :as flow}]
+   (when-not flow-id (throw (ex-info "flow-id is mandatory" flow)))
+
+   (let [flow' (assoc flow :id flow-id)]
```

Comment:
Looking at this it occurred to me if it might be worth using records for these object shapes?

`(defrecord Flow [id inputs output path])` (and possibly similar with interceptors etc)

In JVM Clojure field access on records is much faster than map lookups, though I don't know if that applies in cljd

Reply:
It would be interesting to do some performance bench-marking at some point, and compare between using maps vs  records.

My view is that it would need to be a significant, measurable, performance gain before we swap the simplicity of maps for records

Reply:
I don't feel records would make anything that much more complicated though.

In Clojure they're just glorified maps anyway